### DISCUSSION TITLE
Spatial structure inconsistencies with IFC4x3

### DISCUSSION THREAD

**Discussion by RickBrice** [jwouellette/TPF-5_372-Unit_Test_Suite#9] [Q&A] [answered]
My understanding of Option 1c is that IfcBridgePart::Foundation is IfcRelContainedInSpatialStructure of IfcBridgePart::Pier and IfcBridgePart::Pier is IfcRelContainedInSpatialStructure of IfcBridgePier::Substructure.

This construct is not consistent with IFC4x3. Specifically, 5.4.3.52.4 WR31 and 5.4.3.68.4 WR 41 are violated.

These violations are identified by the bSI validation service.

Is this being reconciled with bSI?

**Comment by jwouellette** [comment]
I don't know , @RickBrice. I sent documentation of our options to bSI (Evandro) when we made the proposals and never heard back from them.

**Comment by SergejMuhic** [comment]
@RickBrice my understanding is that those are `IfcRelAggregates`. `IfcRelContainedInSpatialStructure` is used only for `IfcElement`.

At least that is how I modelled the requirements in the test instructions for IFC Bridge.

Here a screenshot:
![image](https://github.com/jwouellette/TPF-5_372-Unit_Test_Suite/assets/16683597/6c22ced9-57d3-49c2-8cba-51bc2f91c4e7)
It is all Spatial De-/Composition (4.1.4.1.4 and 4.1.4.1.5).

**Comment by TLiebich** [comment, marked as answer]
Facility and facility part (such as IfcBridge and IfcBridgePart) are spatial structure elements, similar to IfcBuilding and IfcBuildingStorey. The aggregation relationship among them is IfcRelAggregates.

Assigning elements (subtypes of IfcElement) to such spatial structure elements, is handled by IfcRelContaintedInSpatialStructure. This had been established long time ago in IFC2x in the building domain and had been reused in IFC4.3 also for Infrastructure.

**Reply by RickBrice** [reply]
@TLiebich I think I misunderstood the IFC on this. There is still much for me to learn. I've changed my software to use the correct spatial structure and it passes the validation service checks.

**Comment by jwouellette** [comment]
The proposal indicates that Superstructure, Substructure, Deck, Pier, Foundation within Pier, Abutment, and Foundation within Abutment are all Spatial Structures (IfcBridgePart). Our understanding is that it is valid to nest spatial structure elements via IfcRelAggregates (e.g. Bridge>BridgePart.Substructure>BridgePart.Pier>BridgePart.Foundation)

The reason we did this was because a "Pier" is not a simple object or IfcElementAssembly, as there are many other elements/concepts that comprise the totality of a "Pier" in its many possible permutations and each have important semantic data and relationships that we want to maintain. We're trying to be consistent in how the US market stakeholders looks at a bridge and its components.

The exceptions we made to this are Approach. Slope Protection, and Retaining Wall since they have a more... nebulous... relationship to the bridge spatial and functional hierarchies as not everyone agrees "where" they are relatively placed to everything else (e.g., Bridge? BridgePart.Substructure? BridgePart.Abutment?), as well as being relatively simple.

Does this work from the vendors' points of view?

**Comment by SergejMuhic** [comment]
Just out of curiosity, is there something wrong with my answer?
